raft.go

```diff
@@ -977,41 +975,49 @@ func (r *Raft) runLeader() {
 	r.leaderLoop()
 }
 
-// startReplication is a helper to setup state and start async replication to a peer.
-// It's safe to call this with the local server (in that case, it does nothing.)
-func (r *Raft) startReplication(server Server) {
-	if server.ID == r.localID {
-		return
-	}
+// startStopReplication will set up state and start asynchronous replication to
+// new peers, and stop replication to removed peers. Before removing a peer,
+// it'll instruct the replication routines to try to replicate to the current
+// index.
+func (r *Raft) startStopReplication() {
+	inConfig := make(map[ServerID]bool, len(r.leaderState.replState))
```

Comment:
This should probably be `len(r.configurations.latest.Servers)`. This'll be 0 at this time.